test/integration/forms/form.js

```diff
@@ -0,0 +1,345 @@
+              template: '<span>{{ item }}</span>',
+              refreshOn: 'data',
+              key: 'select',
+              type: 'select',
+              input: true,
+            },
+            {
+              label: 'Select Bar',
+              widget: 'choicesjs',
+              tableView: true,
+              dataSrc: 'custom',
+              data: {
+                custom: 'values = getDirectory(\'bar\', { filter: { foo: \'bar\' }})',
+              },
+              template: '<span>{{ item }}</span>',
+              refreshOn: 'data',
+              key: 'select2',
+              type: 'select',
+              input: true,
+            },
+          ],
+        };
+      })
+      .routeFormActionFields()
+      .routeFormResponse(fx => {
+        fx.data.storyTime = 'Once upon a time...';
+
+        return fx;
+      })
+      .routeActionActivity()
+      .routePatientByAction(fx => {
+        fx.data.attributes.first_name = 'Testin';
+
+        return fx;
+      })
+      .visit('/patient-action/1/form/11111')
+      .wait('@routeFormByAction')
+      .wait('@routeAction')
+      .wait('@routePatientByAction')
+      .wait('@routeFormDefinition');
+
+    cy
+      .wait('@routeDirectoryBar')
+      .wait('@routeDirectoryFoo')
+      .itsUrl()
+      .should(({ search, pathname }) => {
+        expect(search).to.contain('?filter[foo]=bar');
+        expect(pathname).to.equal('/api/directory/foo');
+      });
+
+    cy
+      .iframe()
+      .find('.formio-component-select .dropdown')
+      .first()
+      .click();
+
+    cy
+      .iframe()
+      .find('.choices__list--dropdown.is-active')
+      .find('.choices__item--selectable')
+      .first()
+      .should('contain', 'one')
+      .next()
+      .should('contain', 'two')
+      .click();
+
+    cy
+      .iframe()
+      .find('.formio-component-select .dropdown')
+      .last()
+      .click();
+
+    cy
+      .iframe()
+      .find('.choices__list--dropdown.is-active')
+      .find('.choices__item--selectable')
+      .first()
+      .should('contain', 'bar')
+      .next()
+      .should('contain', 'baz');
+  });
+
+  specify('update patient field', function() {
+    cy
+      .route({
+        method: 'PATCH',
+        url: '/api/patients/1/fields/foo',
+        response: { data: { attributes: { name: 'foo', value: ['one', 'two'] } } },
+      })
+      .as('routePatchPatientFieldFoo')
+      .route({
+        method: 'PATCH',
+        url: '/api/patients/1/fields/bar',
+        response: { data: { attributes: { name: 'bar', value: ['bar', 'baz'] } } },
+      })
+      .as('routePatchPatientFieldBar')
+      .routeFormDefinition(fx => {
+        return {
+          display: 'form',
+          components: [
+            {
+              label: 'Select Foo',
+              widget: 'choicesjs',
+              tableView: true,
+              dataSrc: 'custom',
+              data: {
+                custom: 'values = updateField(\'foo\', [\'one\', \'two\'])',
```

Comment:
this would not be a likely place within a form to use this API, but it does work to hit the patch and test that values are returned.